plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -935,17 +935,17 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta interface
 	if node == nil {
 		return false, nil, fmt.Errorf("node not found")
 	}
-	if !c.satisfiesExistingPodsAntiAffinity(pod, meta, nodeInfo) {
-		return false, []algorithm.PredicateFailureReason{ErrPodAffinityNotMatch}, nil
+	if result, err := c.satisfiesExistingPodsAntiAffinity(pod, meta, nodeInfo); !result {
+		return false, []algorithm.PredicateFailureReason{ErrPodAffinityNotMatch}, err
```

Comment:
We should not return these as errors. We define three new predicate failure reasons and return them in `[]algorithm.PredicateFailureReason` in addition to `ErrPodAffinityNotMatch`.

Reply:
oh, misunderstanding.:(
Will do that.

Reply:
yes, you are right. I make a misunderstanding. I have made another modification.